packages/foam-vscode/src/features/open-dated-note.ts

```diff
@@ -0,0 +1,205 @@
+import {
+  workspace,
+  ExtensionContext,
+  commands,
+  languages,
+  CompletionItemProvider,
+  CompletionItem,
+  CompletionItemKind,
+  CompletionList
+} from "vscode";
+import { getDailyNoteFileName, openDailyNoteFor } from "../dated-notes";
+import { LinkReferenceDefinitionsSetting } from "../settings";
+import { FoamFeature } from "../types";
+
+interface DateSnippet {
+  snippet: string;
+  date: Date;
+  detail: string;
+}
+
+const daysOfWeek = [
+  { day: "sunday", index: 0 },
+  { day: "monday", index: 1 },
+  { day: "tuesday", index: 2 },
+  { day: "wednesday", index: 3 },
+  { day: "thursday", index: 4 },
+  { day: "friday", index: 5 },
+  { day: "saturday", index: 6 }
+];
+
+const foamConfig = workspace.getConfiguration("foam");
+const foamExtension = foamConfig.get("openDailyNote.fileExtension");
+const foamLinkReferenceDefinitions = foamConfig.get(
+  "edit.linkReferenceDefinitions"
+);
+
+const generateDayOfWeekSnippets = (): DateSnippet[] => {
+  const getTarget = (day: number) => {
+    const target = new Date();
+    const currentDay = target.getDay();
+    const distance = (day + 7 - currentDay) % 7;
+    target.setDate(target.getDate() + distance);
+    return target;
+  };
+  const snippets = daysOfWeek.map(({ day, index }) => {
+    const target = getTarget(index);
+    return {
+      date: target,
+      detail: `Get a daily note link for ${day}`,
+      snippet: `/${day}`
+    };
+  });
+  return snippets;
+};
+
+const createCompletionItem = ({ snippet, date, detail }: DateSnippet) => {
+  const completionItem = new CompletionItem(
+    snippet,
+    CompletionItemKind.Snippet
+  );
+  completionItem.insertText = getDailyNoteLink(date);
+  completionItem.detail = `${completionItem.insertText} - ${detail}`;
+  completionItem.command = {
```

Comment:
do I understand correctly that this will also automatically open the note?
will it be possible to add the link without navigating to the note?

Reply:
Yes it will.

It makes a good candidate for a config option, but perhaps that's best saved for a follow up PR? Up to you, I'm happy to address it as it shouldn't be too long to add in.

Reply:
Intuitively I would feel that the snippet shouldn't navigate away automatically for a couple of reasons (both very common e.g. in my usage):
- I mainly use the snippet to not worry about the daily note format and have shortcuts for text expansion
- I don't always/necessarily want to create an actual note, an "empty" placeholder is usually enough (so that in the graph things that happened in the same day are connected)
- I can still easily navigate through the link by cmd+click (or F12)
- I could go back to the original file, but it feels it interferes with the writing flow


My suggestion would be:
- by default only expand the text without navigating
- have a configuration option to add the navigation
- (not sure if possible, and probably wouldn't worry about this in this PR unless trivial) have a key modifier on the snippet (e.g. cmd+enter) to expand + navigate

wdyt?

Reply:
Yeah, you're right. I think because I wanted to validate the file destination was correct I ended up with this behaviour 😆

I've pushed a new commit which addresses this now. If you'd like the settings name changed lmk and I'll get to it!
The key modifier doesn't _seem_ possible, but defo one to investigate for a separate PR